docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,163 @@
+   
+   ```
+       "MCU_NRF52840": {
+           "inherits": ["Target"],
+           "components_add": […],
+           "core": "Cortex-M4F",
+           "macros": […],
+           "features": […],
+           "device_has": […],
+           "extra_labels": […],
+           "config": {…},
+           "overrides": {…},
+           "OUTPUT_EXT": "hex",
+           "supported_toolchains": ["GCC_ARM", "ARM", "IAR"],
+           "public": false,
+           "detect_code": ["1101"],
+           "bootloader_supported": true
+       }
+   ```
+   
+   For reference, the standard NRF52840 development kit configuration is below. It inherits from the MCU target.
+   
+   ```
+       "NRF52840_DK": {
+           "supported_form_factors": ["ARDUINO"],
+           "inherits": ["MCU_NRF52840"],
+           "release_versions": ["5"],
+           "device_name": "nRF52840_xxAA"
+       },
+   ```
+   
+   In this example, the new board is similar to the NRF52840 development kit. 
+   
+1. Instead of making edits within `mbed-os`, create a `custom_target.json` file, and put it in a directory alongside `mbed-os`.
+   
+   For example:  
+   
+   ```
+   {
+   "IMAGINARYTARGET": {
+           "inherits": ["MCU_NRF52840"],
+           "release_versions": ["5"],
+           "device_name": "nRF52840_xxAA",
+           "detect_code": ["1234"],
+           "device_has_remove": ["ANALOGIN"],
+           “components_remove”: ["QSPI"]
+       }
+   }
+   ```
+   
+   The new board target configuration inherits the MCU target, then follows similar configurations to the development kit, including specifying it should build with Mbed OS 5. Also, it uses `device_name nRF52840_xxAA`, which matches the CMSIS pack for that device. The `detect_code` is a unique ID that identifies the board to the Mbed OS test tools.
+   
+   <span class="notes">**Note:** Contact the Arm Mbed team to get a unique ID, especially if you plan to have your board coexist with other Mbed Enabled boards while running automated tests.</span>
+
+1. The new ImaginaryBoard board does not use AnalogIn, and it does not have a Quad SPI. Remove those. 
+
+   1. AnalogIn is an Mbed OS HAL driver, which the MCU target added using `device_has`. Remove it with `device_has_remove`. 
+   1. Similarly, the MCU target added QSPI using `components_add`. Remove it with `components_remove`.
+
+1. Add `device_has_remove` to disassociate the AnalogIn driver from your target.
+
+1. (Optional) You can also use `device_has_add` to add additional drivers.
+
+   <span class="notes">**Note:** If you choose to add a driver, you may have to provide the driver implementation if it is not already available for your hardware.</span>
+
+1. (Optional) Similarly, you can use `features_add`, `features_remove`, `components_add`, `components_remove`, `macros_add` and `macros_remove`.
+
+## Configuring the code directory and folders
+
+The target source code directory should follow a similar structure as the target configuration.
```

Comment:
You don't tell me where that directory goes (although I can see in the code block that it goes in mbed-os, it's not obvious that it's mandatory because of how the "For example" sentence is structured)